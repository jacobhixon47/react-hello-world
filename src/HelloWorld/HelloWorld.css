.hello-world {
  margin: 10px;
  padding-top: 20px;
  padding-bottom: 20px;
  width: 50%;
  flex-grow: 1;
  text-align: center;
  background-color: #223454;
  color: #dddddd;
}

.spanishify, .englishify, .frenchify, .remove {
  padding: 2%;
  border: none;
  width: 15%;
  margin: 5px;
  border-radius: 20px;
  color: #eeeeee;
}

.spanishify:hover, .englishify:hover, .frenchify:hover {
  background-color: #aaaaaa;
  cursor: pointer;
}

.spanishify {
  background-color: #553FFF;
}

.englishify {
  background-color: #009922;
}

.frenchify, .remove {
  background-color: #770F40;
}
