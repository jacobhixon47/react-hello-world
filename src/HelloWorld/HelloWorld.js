import React, { Component } from 'react';
import './HelloWorld.css';

class HelloWorld extends Component {
  constructor(props) {
    super(props);
    this.state = {
      greeting: 'Hello'
    };
    this.frenchify = this.frenchify.bind(this);
    this.englishify = this.englishify.bind(this);
    this.spanishify = this.spanishify.bind(this);
    this.removeGreeting = this.removeGreeting.bind(this);
  }
  frenchify() {
    this.setState({ greeting: 'Bonjour' });
  }
  spanishify() {
    this.setState({ greeting: 'Hola' });
  }
  englishify() {
    this.setState({ greeting: 'Hello' });
  }
  removeGreeting() {
    this.props.removeGreeting(this.props.name);
  }
  render() {
    return (
      <div className="hello-world">
        {this.state.greeting}, {this.props.name}!
        <br/>
        <button className="frenchify" onClick={this.frenchify}>Frenchify!</button>
        <br/>
        <button className="spanishify" onClick={this.spanishify}>Spanishify!</button>
        <br/>
        <button className="englishify" onClick={this.englishify}>Englishify!</button>
        <br/>
        <button className="remove" onClick={this.removeGreeting}>Remove Me!</button>
      </div>
    );
  }
}

export default HelloWorld;
