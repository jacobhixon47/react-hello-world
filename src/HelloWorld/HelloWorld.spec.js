import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import HelloWorld from './HelloWorld';

describe(HelloWorld, () => {
  // tests
  // generic snapshot test
  it ('renders and matches the snapshot', () => {
    const component = renderer.create(
      <HelloWorld name='Person' />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
// vars for feature tests
  const name = 'Person';
  // fake a removeGreeting method
  const mockRemoveGreeting = jest.fn();
  // fake 'shallow' component
  const component = shallow(
    <HelloWorld name={name} removeGreeting={mockRemoveGreeting} />
  );
  // test that component contains name => 'Person'
  it('contains the supplied name', () => {
    expect(component.text()).toContain(name);
  });
// check translate buttons
  it('modifies the greeting when frenchify/spanishify/englishify button is clicked', () => {
    // french button
    component.find('button.frenchify').simulate('click');
    expect(component.text()).toContain('Bonjour');
    // spanish button
    component.find('button.spanishify').simulate('click');
    expect(component.text()).toContain('Hola');
    // english button
    component.find('button.englishify').simulate('click');
    expect(component.text()).toContain('Hello');
  });
  // check remove button
  it('calls the passed-in removeGreeting when remove button is clicked', () => {
    component.find('button.remove').simulate('click');
    expect(mockRemoveGreeting).toBeCalled();
  });
});
