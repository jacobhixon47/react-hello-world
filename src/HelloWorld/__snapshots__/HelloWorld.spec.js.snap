// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class HelloWorld extends _react.Component {
  constructor(props) {
    super(props);
    this.state = {
      greeting: 'Hello' };

    this.frenchify = this.frenchify.bind(this);
    this.englishify = this.englishify.bind(this);
    this.spanishify = this.spanishify.bind(this);
    this.removeGreeting = this.removeGreeting.bind(this);
  }
  frenchify() {
    this.setState({ greeting: 'Bonjour' });
  }
  spanishify() {
    this.setState({ greeting: 'Hola' });
  }
  englishify() {
    this.setState({ greeting: 'Hello' });
  }
  removeGreeting() {
    this.props.removeGreeting(this.props.name);
  }
  render() {
    return (
      _react2.default.createElement('div', { className: 'hello-world', __source: { fileName: _jsxFileName, lineNumber: 29 }, __self: this },
        this.state.greeting, ', ', this.props.name, '!',
        _react2.default.createElement('br', { __source: { fileName: _jsxFileName, lineNumber: 31 }, __self: this }),
        _react2.default.createElement('button', { className: 'frenchify', onClick: this.frenchify, __source: { fileName: _jsxFileName, lineNumber: 32 }, __self: this }, 'Frenchify!'),
        _react2.default.createElement('br', { __source: { fileName: _jsxFileName, lineNumber: 33 }, __self: this }),
        _react2.default.createElement('button', { className: 'spanishify', onClick: this.spanishify, __source: { fileName: _jsxFileName, lineNumber: 34 }, __self: this }, 'Spanishify!'),
        _react2.default.createElement('br', { __source: { fileName: _jsxFileName, lineNumber: 35 }, __self: this }),
        _react2.default.createElement('button', { className: 'englishify', onClick: this.englishify, __source: { fileName: _jsxFileName, lineNumber: 36 }, __self: this }, 'Englishify!'),
        _react2.default.createElement('br', { __source: { fileName: _jsxFileName, lineNumber: 37 }, __self: this }),
        _react2.default.createElement('button', { className: 'remove', onClick: this.removeGreeting, __source: { fileName: _jsxFileName, lineNumber: 38 }, __self: this }, 'Remove Me!')));


  }} renders and matches the snapshot 1`] = `
<div
  className="hello-world"
>
  Hello
  , 
  Person
  !
  <br />
  <button
    className="frenchify"
    onClick={[Function]}
  >
    Frenchify!
  </button>
  <br />
  <button
    className="spanishify"
    onClick={[Function]}
  >
    Spanishify!
  </button>
  <br />
  <button
    className="englishify"
    onClick={[Function]}
  >
    Englishify!
  </button>
  <br />
  <button
    className="remove"
    onClick={[Function]}
  >
    Remove Me!
  </button>
</div>
`;
